packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -1172,6 +1172,15 @@ export class PipelineEditor extends React.Component<
     ) {
       validationErrors.push('no runtime image');
     }
+    if (node.app_data.cpu <= 0) {
+      validationErrors.push('CPU must be greater than 0');
+    }
+    if (node.app_data.gpu <= 0) {
```

Comment:
Seems like 0 GPU should be allowed.  The same might go for CPU, although I don't really know if something that uses GPUs can exclude the use of CPUs (I kinda doubt it).

Reply:
oh yea that makes sense, I was just going off of what the ui controls were validating for, I'll update those too

so, like this?
cpu > 0
memory > 0 
gpu >= 0

Reply:
Those make sense to me - thanks.